compiler/src/iree/compiler/Dialect/Flow/Transforms/test/hoist_encoding_ops.mlir

```diff
@@ -0,0 +1,158 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(util.func(iree-flow-hoist-encoding-ops))" --split-input-file %s | FileCheck %s
+
+#map1 = affine_map<(d0, d1, d2, d3) -> (d0, d3, d2)>
+#map2 = affine_map<(d0, d1, d2, d3) -> (d0, d1, d3)>
+#map3 = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2)>
+module {
+  util.func public @hoist_matmul_encodings(%arg0: tensor<2x128x64xf32>, %arg1: tensor<2x11008x128xf32>) -> tensor<2x11008x64xf32> {
+    %cst = arith.constant 0.000000e+00 : f32
+    %2 = flow.dispatch.region -> (tensor<2x11008x64xf32>) {
+      %3 = iree_encoding.set_encoding %arg0 : tensor<2x128x64xf32> -> tensor<2x128x64xf32, #iree_encoding.encoding<operand_index = 0 : index, op_type =  matmul, element_types = [f32, f32, f32], original_type = tensor<2x128x64xf32>, user_indexing_maps = [#map1, #map2, #map3], round_dims_to = array<i64: 32, 32, 32>>>
```

Comment:
Can you do something similar to https://github.com/iree-org/iree/pull/18069/commits/0e19ef3c6e4f5f954e059db5d05c4b68d7f3616a# that defines `#encoding` variables and use them in IRs? It makes it much easier to read.